db/version_edit_handler.cc

```diff
@@ -111,6 +111,14 @@ Status FileChecksumRetriever::ApplyVersionEdit(VersionEdit& edit,
       return s;
     }
   }
+  for (const auto& new_blob_file : edit.GetBlobFileAdditions()) {
+    Status s = file_checksum_list_.InsertOneFileChecksum(
+        new_blob_file.GetBlobFileNumber(), new_blob_file.GetChecksumValue(),
+        new_blob_file.GetChecksumMethod());
```

Comment:
In general, we also add files without checksums to the list, right?

If so, there's one difference between how this case is handled for table files and blob files. IIRC for table files, `file_checksum_func_name` would contain `kUnknownFileChecksumFuncName`, while for blob files it would be empty. It would be nice to make these uniform when adding them to `file_checksum_list_` by replacing an empty checksum method with `kUnknownFileChecksumFuncName` when adding blob files.

Reply:
Oh. Thanks for pointing it out. Also is it possible for some files to have checksums and some not from testing purpose if i want to add a unit test. Is it possible through an option in general?

Reply:
> Also is it possible for some files to have checksums and some not from testing purpose if i want to add a unit test. Is it possible through an option in general?

In general, it's possible: one can open a database without specifying `file_checksum_gen_factory`, close it, then reopen it with `file_checksum_gen_factory` set. (It cannot be adjusted on the fly though while the DB is open b/c it's an immutable option.)

Reply:
Sorry, what I meant here is that we could translate an empty checksum method/value for blob files to `kUnknownFileChecksumFuncName` / `kUnknownFileChecksum` when we build the `FileChecksumList` in `FileChecksumRetriever`.  Changing the MANIFEST related stuff (`BlobFileAddition` and friends) does not seem backwards compatible.

Reply:
oh ok. By adding blob files I thought when blob files are added to db.  I will update the code.